src/chips/LMS7002M/MCU_File.cpp

```diff
@@ -425,32 +426,25 @@ void MCU_File::ReadHex(unsigned long limit)
     {
         throw std::runtime_error("No data in file!"s);
     }
-    vector<MemBlock>::iterator vi;
-    m_top = 0;
-    for (vi = m_chunks.begin(); vi < m_chunks.end(); vi++)
-    {
-        m_top = std::max(m_top, (vi->m_startAddress + vi->m_bytes.size() - 1));
```

Comment:
It is a max value accumulation, but since it is not a straight forward maximum element search (which could easily be done with `std::max_element`), using this, I feel like, is the cleanest solution.

Code comparison between the original, current and a third, potential idea:
https://godbolt.org/z/8sr8qr3MP

Reply:
There is no need to overcomplicate things and use every available feature of the language.
A simple loop is far more comprehensible and readable than the lambda monstrosities.
```
struct MemBlock {
    size_t m_startAddress;
    std::vector<unsigned char> m_bytes;
    size_t GetEndAddress() const { return m_startAddress + m_bytes.size() - 1; }
};

m_highestAddress = 0;
for (const auto& chunk: m_chunks)
    m_highestAddress = std::max(m_highestAddress, chunk.GetEndAddress());
```